### PR TITLE
backend: add preference spot implementation and tests

### LINKED ISSUE #139: Database: Parking spot preference
We need to let the booker store preference parking spot for ease of access. This needs a table to facilitate the connection between User and Parking spot. This should be simple I think
Refer to #29

### PR DESCRIPTION
**Pull request by JZ-FSDev**
Implemented the following endpoints along with tests:

- POST `/spots/{id}/preference`: Creates the preference
- GET `/spots/{id}/preference`: Gets a boolean result indicating if the parking spot is preferred by the user
- GET `/spots/preference`: Gets a list of all parking spots that are preferred by the user with pagination
- DELETE `/spots/{id}/preference`: Deletes the preference

Closes #139 #142

### CODE REVIEW

File: backend/internal/pkg/dbmigration/migrations/000009_create_preferencespot_table.up.sql
[the remark this answers is not in the document]
Comment (JZ-FSDev) [reply backend/internal/pkg/dbmigration/migrations/000009_create_preferencespot_table.up.sql:1]: Thanks for noticing.  I had the expectation that booking would be completed before my PR goes through so the numbers would have aligned naturally, but alas.

File: backend/internal/pkg/dbmigration/migrations/000008_create_preferencespot_table.up.sql [hunk 1]
```diff
@@ -0,0 +1,7 @@
+CREATE TABLE IF NOT EXISTS PreferenceSpot (
+    PreferenceSpotID BIGSERIAL PRIMARY KEY,
+    UserID BIGINT NOT NULL,
```
[the remark this answers is not in the document]
Comment (JZ-FSDev) [reply backend/internal/pkg/dbmigration/migrations/000008_create_preferencespot_table.up.sql:3]: Yes, I will add it back.  I was doing some tests when testing the queries and suspected the constraint was causing problems.  Though I am technically not ever joining user table for preference spots, having the constraint is still good for integrity.  Thanks for pointing it out.

**Review by HritikPunj — DISMISSED** [review]
Thanks and great work!

**Comment by JZ-FSDev** [comment]
Thanks for the review Hieu.  I chose not to refactor the naming from preference to preferred since it sounds odd to me though I agree with your point about the adjective modifying the spot.  The main thing is that it is not a verb and I would rather focus time on completing the other features and refactor naming and additional adjustments in a separate PR.  All other items pointed out I have addressed.

**Review by HritikPunj — APPROVED** [review]
Thanks and great job

**Pull request merged**